common/sgx/collateral.c

```diff
@@ -341,44 +338,45 @@ oe_result_t oe_validate_revocation_list(
          */
         else
         {
-            der_data = sgx_endorsements->items[i].data;
-            der_data_size = sgx_endorsements->items[i].size;
+            uint8_t* der_data = sgx_endorsements->items[i].data;
+            size_t der_data_size = sgx_endorsements->items[i].size;
             if (der_data_size == 0 || der_data == NULL)
                 OE_RAISE(OE_INVALID_PARAMETER);
 
+            // If CRL buffer has null terminator, remove it.
+            if (der_data[der_data_size - 1] == 0)
+                der_data_size -= 1;
+
             // Check if the CRL is composed of only hex digits
-            for (size_t l = 0; l < der_data_size; l++)
+            bool ishex = der_data_size % 2 == 0;
+            if (ishex)
             {
-                const uint8_t c = sgx_endorsements->items[i].data[l];
-                if (!isxdigit(c))
+                for (size_t l = 0; l < der_data_size; l++)
                 {
-                    ishex = false;
-                    break;
+                    if (!isxdigit(der_data[l]))
+                    {
+                        ishex = false;
+                        break;
+                    }
                 }
             }
 
             // If CRL is a hex string, convert hex to der
             if (ishex)
             {
-                der_data_size = (der_data_size / 2) + (der_data_size % 2);
-                der_data = oe_malloc(der_data_size);
+                const char* const hex_data = (char*)der_data;
+                const size_t hex_data_size = der_data_size;
+                der_data_size /= 2;
```

Comment:
Do we not need the der_data_size%2 addition for odd numbers here?

Reply:
I changed it so that we only try to interpret even length data as hex:
https://github.com/openenclave/openenclave/blob/01edb4cbf087ce3f0759b94d2ce3b780fc773aaa/common/sgx/collateral.c#L351-L358
So at this point size is always even.